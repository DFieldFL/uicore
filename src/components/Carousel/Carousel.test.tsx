import React from 'react'
import { render } from '@testing-library/react'
import { Carousel } from './Carousel'

const DummyComponent = () => {
  return (
    <Carousel>
      <div style={{ backgroundColor: 'red', height: 400, color: 'white', textAlign: 'center' }}>Image 1</div>
      <div style={{ backgroundColor: 'blue', height: 400, color: 'white', textAlign: 'center' }}>Image 2</div>
      <div style={{ backgroundColor: 'green', height: 400, color: 'white', textAlign: 'center' }}>Image 3</div>
      <div style={{ backgroundColor: 'red', height: 400, color: 'white', textAlign: 'center' }}>Image 4</div>
      <div style={{ backgroundColor: 'blue', height: 400, color: 'white', textAlign: 'center' }}>Image 5</div>
      <div style={{ backgroundColor: 'green', height: 400, color: 'white', textAlign: 'center' }}>Image 6</div>
    </Carousel>
  )
}

describe('Render basic component', () => {
  test('should check snapshot', () => {
    const { container } = render(<DummyComponent />)
    expect(container).toMatchSnapshot()
  })
})
