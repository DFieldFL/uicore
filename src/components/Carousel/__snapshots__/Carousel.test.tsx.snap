// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render basic component should check snapshot 1`] = `
<div>
  <div
    class="vertical layout-spacing-xsmall main carousel spacing spacing-xsmall padding padding-small"
  >
    <div
      class="carouselView"
    >
      <span>
         
      </span>
      <div
        class="carouselItem active"
      >
        <div
          style="background-color: red; height: 400px; color: white; text-align: center;"
        >
          Image 1
        </div>
      </div>
      <div
        class="carouselItem carouselItemNext"
      >
        <div
          style="background-color: blue; height: 400px; color: white; text-align: center;"
        >
          Image 2
        </div>
      </div>
      <div
        class="carouselItem"
      >
        <div
          style="background-color: green; height: 400px; color: white; text-align: center;"
        >
          Image 3
        </div>
      </div>
      <div
        class="carouselItem"
      >
        <div
          style="background-color: red; height: 400px; color: white; text-align: center;"
        >
          Image 4
        </div>
      </div>
      <div
        class="carouselItem"
      >
        <div
          style="background-color: blue; height: 400px; color: white; text-align: center;"
        >
          Image 5
        </div>
      </div>
      <div
        class="carouselItem"
      >
        <div
          style="background-color: green; height: 400px; color: white; text-align: center;"
        >
          Image 6
        </div>
      </div>
      <div
        class="carouselRight"
      >
        <div>
          <span
            class="bp3-icon bp3-icon-circle-arrow-right main color color-white"
            icon="circle-arrow-right"
          >
            <svg
              data-icon="circle-arrow-right"
              height="30"
              viewBox="0 0 20 20"
              width="30"
            >
              <desc>
                circle-arrow-right
              </desc>
              <path
                d="M15.71 9.29l-4-4a1.003 1.003 0 00-1.42 1.42L12.59 9H5c-.55 0-1 .45-1 1s.45 1 1 1h7.59l-2.29 2.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4-4c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71zM10 0C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <div
      class="carouselIndicators"
    >
      <span
        class="indicator active"
      />
      <span
        class="indicator"
      />
      <span
        class="indicator"
      />
      <span
        class="indicator"
      />
      <span
        class="indicator"
      />
      <span
        class="indicator"
      />
    </div>
  </div>
</div>
`;
